vibes-tools/resources/patch-info.2.json

```diff
@@ -1,5 +1,25 @@
 {
   "patch-point": "0x10400:32",
   "patch-size": 2,
-  "sp-align": 0
+  "sp-align": 0,
+  "patch-vars": [
+    {
+      "name": "x",
+      "storage-class": [
+        "register",
+        {
+          "allow-opt": true
```

Comment:
I'm behind the times. What does "allow-opt" do?

Reply:
The docstring in `vibes-higher-vars/higher_var.mli` explains it. Basically when we run BAP's BIR optimization, it will do dead code elimination and constant propagation on variables. By default it will only do this on virtual variables (since these didn't exist in the original patch program). However, we can "opt-in" for it to do this on higher variables in the program. This is off by default because generally we write micro-patches like:

```
int x;
x = 5;
```

where we haven't really specified what register `x` should go in, so it would result in an empty patch if we optimized it by default.

Reply:
Most of the time, this feature shouldn't be needed because (in a perfect world) the register allocator would pick assignments that result in the fewest registers used, resulting in code like so:

```
mov R0, R0
add R0, #1
```

which is then passed to the peephole optimizer.